zookeeper-api/src/main/java/org/apache/helix/zookeeper/zkclient/callback/ZkAsyncCallbacks.java

```diff
@@ -50,6 +50,10 @@ public void processResult(int rc, String path, Object ctx, byte[] data, Stat sta
         if (_data != null && ctx != null && ctx instanceof ZkAsyncCallMonitorContext) {
           ((ZkAsyncCallMonitorContext) ctx).setBytes(_data.length);
         }
+      } else if(rc != Code.NONODE.intValue()) {
+        if (ctx instanceof ZkAsyncCallMonitorContext) {
+          ((ZkAsyncCallMonitorContext) ctx).recordFailure(path);
+        }
       }
       callback(rc, path, ctx);
```

Comment:
Why not process the result in the callback method? So that you process all the results in one place. The current version only records failures of 2 out of 5 zk calls.

Reply:
Thank you, @jiajunwang, your suggestion makes more sense. I will use it in the next commit.